gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -188,10 +214,17 @@ private ManagedChannel createSingleChannel() throws IOException {
     int port = Integer.parseInt(endpoint.substring(colon + 1));
     String serviceAddress = endpoint.substring(0, colon);
 
-    // After CallCredentials is supported, change this back to GoogleDefaultChannelBuilder.
-    // https://github.com/googleapis/gax-java/issues/649.
-    ManagedChannelBuilder builder =
-        ManagedChannelBuilder.forAddress(serviceAddress, port)
+    ManagedChannelBuilder builder;
+
+    // TODO(weiranf): Add a new API in ComputeEngineCredentials to check whether it's using default
```

Comment:
Is only the default credentials allowed to use alts?

Reply:
Yes, it's only safe to attempt directpath if the access token used on the channel authenticates VM's default service account, which is the identity authenticated by ALTS connection